plugins/finder/jevents/jevents.php

```diff
@@ -384,19 +342,121 @@ protected function getItem($id)
 	 */
 	protected function getStateQuery()
 	{
-		$sql = $this->db->getQuery(true);
+		$db     = $this->db;
+		$query  = $db->getQuery(true);
+
 		// Item ID
-		$sql->select('a.evdet_id as id');
+		$query->select('a.ev_id');
+
```

Comment:
I was conflicted on #__jevents_vevdetail and #__jevents_vevent because both have state column and which is being updated when doing state change (publish, unpublish, trash). I tested change state on one of the event and it surely does update the _vevent table and _vevdetail state column remains always 1 (as value) regardless of what state I changed it to. So, I use a.ev_id when state is changed.

But I could be wrong...